pulsar-metadata/src/test/java/org/apache/pulsar/metadata/MetadataCacheTest.java

```diff
@@ -331,6 +331,7 @@ public void insertionOutsideCacheWithGenericType(String provider, Supplier<Strin
         });
 
         String key1 = newKey();
+        objCache.get(key1).join();
```

Comment:
This is forcing the key to be in the cache, but the test was expressly trying to check the condition in which it was not.

Reply:
Change the test to `updateOutsideCacheWithGenericType`.
For now, I think if we can implement `insertionOutsideCacheWithGenericType` by make a path pattern registration or add a method like `accept(path)` for each MetadataStore instance, which implies the metadata store could deserialization this value, and the path indeed should be cached in this metadata store.